flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka09PartitionDiscoverer.java

```diff
@@ -102,4 +106,13 @@ protected void closeConnections() throws Exception {
 			this.kafkaConsumer = null;
 		}
 	}
+
+	private void closeConnectionsQuietly() {
+		try {
+			closeConnections();
```

Comment:
I think you need to invoke the based class `AbstractPartitionDiscoverer.close()` to ensure that the state is also set to closed